docs/source/reference-core.rst

```diff
@@ -834,13 +834,13 @@ Nursery objects provide the following interface:
 
       Note that this is *not* an async function and you don't use await
       when calling it. It sets up the new task, but then returns
-      immediately, *before* it has a chance to run. The new task won’t
-      actually get a chance to do anything until some later point when
-      you execute a checkpoint and the scheduler decides to run it.
-      If you want to run a function and immediately wait for its result,
-      then you don't need a nursery; just use ``await async_fn(*args)``.
-      If you want to wait for the task to initialize itself before 
-      continuing, see :meth:`start()`.
+      immediately, *before* it has a chance to run. At some future
+      time, (which might be closing of the nursery context, but no
+      later), all tasks requested with :meth:`start_soon` will begin,
+      in no particular order. If you want to run a function to
+      completion, then you don't need a nursery; just use
```

Comment:
The function always runs "to completion" (as opposed to being abandoned in the middle somewhere) regardless of whether you use `await` or `nursery.start_soon`. What are you trying to communicate here that you didn't feel was adequately communicated with "and immediately wait for its result"?

Reply:
The start function runs a function and immediately waits for a result too. It just so happens that the result might be returned by an intermediate result from a .started call in the middle.. I wanted to convey the difference. How about "If you want to run a function and block until it finishes" ?